### DISCUSSION TITLE
Mocked Zustand: `TypeError: useStore.subscribe is not a function`

### DISCUSSION THREAD

**Discussion by Ademsk1** [pmndrs/zustand#2087] [Q&A] [answered]
Hi there, big fan of the library

Been searching for a while, but can't seem to get the mocked Zustand to work with my tests using React Native. I've added in the `zustand.ts` file in `/__mocks__` using the documentation [here](https://docs.pmnd.rs/zustand/guides/testing). I've been browsing the conversations here and came across [this](https://github.com/pmndrs/zustand/discussions/1546#discussioncomment-4729375) modification of the mocked zustand file, where the extensions of subscribe is used.

**Comment by dbritto-dev** [comment, marked as answer]
@Ademsk1 are you using uncurried version? the current version of testing guide just considered the curried version since that's the preferred way for ts and js. 

In short words you need to use curried version to be able to use our testing guide or update the mock to match the uncurried version.

**Reply by Ademsk1** [reply]
Ah, I'm currently using uncurried, so 

```
useStore = create(...)
```
So is it the case that using the zustand mock is a kind of middleware? 
Also, would be great if this could get a little mention in the docs for the Zustand testing, about paying attention to that.

**Reply by dbritto-dev** [reply]
@Ademsk1 sure I'll update the docs to mention that
